workspace/__lib__/xod/graphics/image/patch.cpp

```diff
@@ -0,0 +1,46 @@
+
+// clang-format off
+{{#global}}
+#include <Image.h>
+{{/global}}
+// clang-format on
+
+struct State {
+    uint8_t mem[sizeof(Image)];
+    Image* image;
+    int16_t x, y, w, h;
+};
+
+// clang-format off
+{{ GENERATED_CODE }}
+// clang-format on
+
+void evaluate(Context ctx) {
+    auto state = getState(ctx);
+
+    auto gfx = getValue<input_GFX>(ctx);
+    auto bmp = getValue<input_BMP>(ctx);
+
+    int16_t x = (int16_t)getValue<input_X>(ctx);
+    int16_t y = (int16_t)getValue<input_Y>(ctx);
+    int16_t w = (int16_t)getValue<input_W>(ctx);
+    int16_t h = (int16_t)getValue<input_H>(ctx);
+
+    if (isSettingUp()) {
+        state->image = new (state->mem) Image(gfx);
+        state->image->setBitmap(bmp.bitmapBuffer, bmp.bitmapWidth, bmp.bitmapHeight);
```

Comment:
What if we want to dynamically replace one image with another one, using `select`/`if-else`/etc node?
For example, imagine that you want to show spinner (or other "loading" picture), while the device connects to the API, and then show icon of the sun/rain/cloud/etc?

Reply:
Maybe extend the evaluate condition?
```
if (isSettingUp() || x != state->x || y != state->y || w != state->w || h != state->h || isInputDirty<input_BMP>(ctx)) {
    state->image->setBitmap(bmp.bitmapBuffer, bmp.bitmapWidth, bmp.bitmapHeight);
}
```